  @Test(expectedExceptions = InfrastructureException.class)
  public void failWhenTryToCreateCmWithWronglySetAnnotation() throws InfrastructureException {
    // given
    when(internalRuntime.getOwner()).thenReturn(OWNER_ID);
    when(internalRuntime.getStatus()).thenReturn(WorkspaceStatus.STARTING);

    ConfigMap cm1 =
        new ConfigMapBuilder()
            .withNewMetadata()
            .withName("cm1")
            .withAnnotations(
                Collections.singletonMap(CREATE_IN_CHE_INSTALLATION_NAMESPACE, "blabol"))
            .endMetadata()
            .build();

    when(clientFactory.create()).thenReturn(kubeClient);
    when(kubeClient.configMaps()).thenReturn(kubeConfigMaps);
    when(kubeConfigMaps.inNamespace(CHE_NAMESPACE)).thenReturn(kubeConfigMapsInNamespace);
    when(kubeConfigMapsInNamespace.create(any(ConfigMap.class))).thenReturn(cm1);

    List<ConfigMap> configMapsToCreate = Collections.singletonList(cm1);

    // when
    cheNamespace.createConfigMaps(configMapsToCreate, identity);

    // then exception
  }

